### DISCUSSION TITLE
https://fuzzylogic.readthedocs.io/en/latest/Discussion/

### DISCUSSION THREAD

**Discussion by giscus** [Python-Fuzzylogic/fuzzylogic#52] [General]
# https://fuzzylogic.readthedocs.io/en/latest/Discussion/

https://fuzzylogic.readthedocs.io/en/latest/Discussion/

**Comment by robyngehler** [comment]
Hello Mr Kiefner,

after installing pip and fuzzylogic on a new set up Manjaro OS I tested some old code of mine, using your fuzzy logic.

I observed an issue in the "__call__" function from the "fuzzylogic/classes.py":
Line 447 "index = sum(v.center_of_gravity * x for v, x in weights) / sum(x for v, x in weights)" threw a TypeError: unsupported operand type(s) for *: 'method' and 'float'

After looking over the code I noticed a missing "()" after the "v.center_of_gravity" function call. With this little change it worked as usual.

I'm not sure why I observed this error just now, maybe it's some bug with installing it on a Manjaro OS, but may you look over it.

Best regards and thanks for this awesome library
Robyn Gehler

**Reply by amogorkon** [reply]
Hi,

thanks for reporting it, it's fixed it in a new commit. Sorry for the inconvenience.

I have been tinkering on the defuzzification algorithm lately. Since it's quite complex I fear there may be more bugs in this part of the lib. If you notice anything spurious, please let me know.

Best regards,
Anselm
